### PR TITLE
chore: adapt apecloud-mysql addon for API 1.0

### PR DESCRIPTION
**Pull request by xuriwuyun**

### CODE REVIEW

File: addons/apecloud-mysql/templates/_helpers.tpl [hunk 1]
```diff
@@ -5,24 +5,6 @@ Expand the name of the chart.
 {{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" }}
 {{- end }}
 
-{{/*
```
Comment (Y-Rookie) [inline addons/apecloud-mysql/templates/_helpers.tpl:8]: remove  all the downwardAPI define in cmpd or _helpers.tpl referencerd by cmpd

File: addons/apecloud-mysql/templates/_helpers.tpl [hunk 2]
```diff
@@ -285,9 +278,28 @@ env:
   - name: KB_MYSQL_CLUSTER_UID
     value: $(KB_CLUSTER_UID)
   - name: KB_MYSQL_N
-    value: $(KB_REPLICA_COUNT)
+    value: $(KB_COMP_REPLICAS)
```
Comment (Y-Rookie) [inline addons/apecloud-mysql/templates/_helpers.tpl:281]: built-in env KB_COMP_REPLICAS will be deleted in the future and please declare it in vars

File: addons/apecloud-mysql/templates/backuppolicytemplate.yaml [hunk 3]
```diff
@@ -7,7 +7,7 @@ metadata:
 spec:
   serviceKind: ApeCloudMySQL
   compDefs:
-   - {{ include "apecloud-mysql.componentDefName" . }}
+   - {{ include "apecloud-mysql.cmpdNameApecloudMySQL" . }}
```
Comment (Y-Rookie) [inline addons/apecloud-mysql/templates/backuppolicytemplate.yaml:10]: here should be match with prefix or regexp pattern

File: addons/apecloud-mysql/templates/backuppolicytemplateforhscale.yaml [hunk 4]
```diff
@@ -7,7 +7,7 @@ metadata:
 spec:
   serviceKind: ApeCloudMySQL
   compDefs:
-    - {{ include "apecloud-mysql.componentDefName" . }}
+    - {{ include "apecloud-mysql.cmpdNameApecloudMySQL" . }}
```
Comment (Y-Rookie) [inline addons/apecloud-mysql/templates/backuppolicytemplateforhscale.yaml:10]: ditto

File: addons/apecloud-mysql/templates/cmpd-wescale-controller.yaml [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+        ports:
+          - containerPort: 15000
+            name: vtctld-webport
+          - containerPort: 15999
+            name: vtctld-grpcport
+        env:
+          - name: CELL
+            value: {{ .Values.wesqlscale.cell | default "zone1" | quote }}
+          - name: VTCTLD_SERVER
+            value: "$(KB_CLUSTER_NAME)-wescale-ctrl-headless"
```
Comment (Y-Rookie) [inline addons/apecloud-mysql/templates/cmpd-wescale-controller.yaml:45]: built-in env KB_CLUSTER_NAME will be removed inthe future too, check them all

File: addons/apecloud-mysql/templates/cmpd-wescale-controller.yaml [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        volumeMounts:
+          - name: scripts
+            mountPath: /scripts
+          - name: data
+            mountPath: /vtdataroot
+  vars:
+    - name: MYSQL_ROOT_USER
+      valueFrom:
+        credentialVarRef:
+          compDef: {{ include "apecloud-mysql.cmpdNameApecloudMySQL" . }}
```
Comment (Y-Rookie) [inline addons/apecloud-mysql/templates/cmpd-wescale-controller.yaml:62]: compdef reference in vars should use prefix or regexp match, check them all

File: addons/apecloud-mysql/templates/_helpers.tpl [hunk 7]
```diff
@@ -285,9 +278,28 @@ env:
   - name: KB_MYSQL_CLUSTER_UID
     value: $(KB_CLUSTER_UID)
```
Comment (Y-Rookie) [inline addons/apecloud-mysql/templates/_helpers.tpl:279]: ditto, KB_CLUSTER_UID will not be suuported in the future

**Review by zjx20 — APPROVED** [review]

**Review by zjx20 — APPROVED** [review]

**Review by Y-Rookie — APPROVED** [review]

**Pull request merged**